index.js

```diff
@@ -62,10 +65,17 @@ module.exports = function(source, map) {
 
 		if (options.emitCss && css.code) {
 			const resource = posixify(compileOptions.filename);
-			const cssPath = `${resource}.${index++}.css`;
+			const threadLoaderUsed = this.emitFile === undefined;
```

Comment:
Perhaps there should be an override to force inline mode to always be enabled or disabled? (this could be written as `options.forceInlineCss ?? this.emitFile === undefined` in Node 14+ but alas versions below that [don't support the `??` operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator))
```suggestion
			const threadLoaderUsed = (options.forceInlineCss === undefined) ? this.emitFile === undefined : options.forceInlineCss;
```

Reply:
It's a reasonable suggestion, but the downside is that the option then becomes part of the API, which makes it hard to change (we'd need another major release). I might wait until a need for it arises to leave us more flexibility instead of adding it up front